lib/node_modules/@stdlib/ndarray/shift/README.md

```diff
@@ -0,0 +1,173 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2025 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# shift
+
+> Return an array containing a read-only truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a read-only view of the first element(s) along a specified dimension.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var shift = require( '@stdlib/ndarray/shift' );
+```
+
+#### shift( x\[, options] )
+
+Returns an array containing a **read-only** truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a **read-only** view of the first element(s) along a specified dimension.
+
+```javascript
+var array = require( '@stdlib/ndarray/array' );
+var ndarray2array = require( '@stdlib/ndarray/to-array' );
+
+var x = array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ], {
+    'shape': [ 3, 2 ]
+});
+// returns <ndarray>
+
+var arr = ndarray2array( x );
+// returns [ [ 1.0, 2.0 ], [ 3.0, 4.0 ], [ 5.0, 6.0 ] ]
+
+var y = shift( x );
+// returns [ <ndarray>, <ndarray> ]
+
+arr = ndarray2array( y[ 0 ] );
+// returns [ [ 3.0, 4.0 ], [ 5.0, 6.0 ] ]
+
+arr = ndarray2array( y[ 1 ] );
+// returns [ [ 1.0, 2.0 ] ]
+```
+
+The function accepts the following arguments:
+
+-   **x**: input ndarray.
+-   **options**: function options (_optional_).
+
+The function supports the following `options`:
+
+-   **dim**: dimension along which to perform the operation. If provided an integer less than zero, the dimension index is resolved relative to the last dimension, with the last dimension corresponding to the value `-1`. Default: `0`.
```

Comment:
The default should be `-1`, not `0`. The nature of the operation does not have any bearing on the `dim` option. Meaning, what value the `dim` option is completely orthogonal to the fact that the function removes the first element(s). We should be consistent with other APIs and having the default dimension be `-1`. That affects the entirety of this PR.